In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv('DATASET\\uk_road_accident.csv')

In [3]:
df

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [4]:
df['Accident Date'] = df['Accident Date'].str.strip()
df['Accident Date'] = df['Accident Date'].astype('str')
df['Accident Date'] = df['Accident Date'].str.replace('/', '-')

In [5]:
df['Accident Date'] = pd.to_datetime(df['Accident Date'], dayfirst = True, errors = 'coerce')

In [6]:
df.dtypes

Index                              object
Accident_Severity                  object
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                   object
District Area                      object
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions            object
Road_Type                          object
Urban_or_Rural_Area                object
Weather_Conditions                 object
Vehicle_Type                       object
dtype: object

In [7]:
df['Year'] = df['Accident Date'].dt.year
df['Month'] = df['Accident Date'].dt.month
df['Day'] = df['Accident Date'].dt.day
df['DayOfWeek'] = df['Accident Date'].dt.dayofweek

<h1>1. How many total accidents are in the data?</h1>

In [8]:
df.shape[0]

660679

<h3>Insight : The total number of accidents in the data is 660,679</h3>

<h1>2. How many accidents happened by severity (Slight, Serious, Fatal)?</h1>

In [9]:
df["Accident_Severity"].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<h3>Insight : Here are the total numbers of accidents by severity: Slight - 563,801, Serious - 88,217, and Fatal - 8,661</h3>

<h1>3. What is the most common accident severity?</h1>

In [10]:
df["Accident_Severity"].mode()[0]

'Slight'

<h3>Insight : The most common accident severity is "Slight"</h3>

<h1>4. On average, how many people are hurt in each accident?</h1>

In [11]:
df["Number_of_Casualties"].mean()

np.float64(1.357040257068864)

<h3>Insight : The average number of casualties per accident is 1.35</h3>

<h1>5. What is the highest number of people hurt in a single accident?</h1>

In [12]:
df["Number_of_Casualties"].max()

np.int64(68)

<h3>Insight : The highest number of people hurt in a single accident is 68</h3>

<h1>6. What is the lowest number of people hurt in a single accident?</h1>

In [13]:
df["Number_of_Casualties"].min()

np.int64(1)

<h3>Insight : The lowest number of people hurt in a single accident is 1</h3>

<h1>7. How many total people were hurt in all accidents?</h1>

In [14]:
df["Number_of_Casualties"].sum()

np.int64(896568)

<h3>Insight : The total people were hurt in all accidents is 896,568</h3>

<h1>8. On average, how many vehicles are involved in one accident?</h1>

In [15]:
df["Number_of_Vehicles"].mean()

np.float64(1.8312554205597575)

<h3>Insight : The average vehicles per accident is 1.83</h3>

<h1>9. What is the highest number of vehicles in one accident?</h1>

In [16]:
df["Number_of_Vehicles"].max()

np.int64(32)

<h3>Insight : The highest number of vehicles in one accident is 32</h3>

<h1>10. What is the most common number of vehicles in an accident?</h1>

In [17]:
df["Number_of_Vehicles"].mode()[0]

np.int64(2)

<h3>Insight : The most common number of vehicles in an accident is 2</h3>

<h1>11. How many accidents happened under each light condition?</h1>

In [18]:
df["Light_Conditions"].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

<h3>Insight : Here are the accidents happened under each light condition:<br>Daylight - 484,880<br>Darkness(lights lit) - 129,335<br>Darkness(no lighting) - 37,437<br>Darkness(lighting unknown) - 6,484<br>Darkness(lights unlit) - 2,543
</h3>

<h1>12. What is the most common light condition during accidents?</h1>

In [19]:
df["Light_Conditions"].mode()[0]

'Daylight'

<h3>Insight : The most common light condition during accidents is Daylight</h3>

<h1>13. How many accidents happened under each weather condition?</h1>

In [20]:
df["Weather_Conditions"].value_counts()

Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<h3>Insight : Here are the accidents happened under each weather condition:<br>Fine no high winds - 520,885<br>Raining no high winds - 79,696<br>Other - 17,150<br>Raining + high winds - 9,615<br>Fine + high winds) - 8,554<br>Snowing no high winds - 6,238<br>Fog or mist - 3,528<br>Snowing + high winds - 8,85
</h3>

<h1>14. What is the most common weather condition during accidents?</h1>

In [21]:
df["Weather_Conditions"].mode()[0]

'Fine no high winds'

<h3>Insight : The most common weather condition during accidents is 'Fine no high winds'
</h3>

<h1>15. Which 5 districts have the most accidents?</h1>

In [22]:
df["District Area"].value_counts().head(5)

District Area
Birmingham    13491
Leeds          8898
Manchester     6720
Bradford       6212
Sheffield      5710
Name: count, dtype: int64

<h3>Insight : The 5 districts that have the most accidents are:<br>Birmingham - 13,491<br>Leeds - 8,898<br>Manchester - 6,720<br>Bradford - 6,212<br>Sheffield - 5,710</h3>

<h1>16. How many accidents happened on each road surface condition?</h1>

In [23]:
df["Road_Surface_Conditions"].value_counts()

Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

<h3>Insight : Here are the accidents that happened on each road surface condition:<br>Dry - 447,821<br>Wet or damp - 186,708<br>Frost or ice - 18,517<br>Snow - 5,890<br>Flood over 3cm. deep - 1,017</h3>

<h1>17. What is the most common road surface condition in accidents?</h1>

In [24]:
df["Road_Surface_Conditions"].mode()[0]

'Dry'

<h3>Insight : The most common road surface condition in accidents is 'Dry'
</h3>

<h1>18. How many accidents happened in urban vs rural areas?</h1>

In [25]:
df["Urban_or_Rural_Area"].value_counts()

Urban_or_Rural_Area
Urban          421663
Rural          238990
Unallocated        11
Name: count, dtype: int64

<h3>Insight : The number of accidents happened in urban vs rural areas are<br>Urban - 421,663<br>Rural - 238,990<br>Unallocated - 11</h3>

<h1>19. Which 5 vehicle types are most often in accidents?</h1>

In [26]:
df["Vehicle_Type"].value_counts().head(5)

Vehicle_Type
Car                                     497992
Van / Goods 3.5 tonnes mgw or under      34160
Bus or coach (17 or more pass seats)     25878
Motorcycle over 500cc                    25657
Goods 7.5 tonnes mgw and over            17307
Name: count, dtype: int64

<h3>Insight : Here are the 5 vehicle types that are most often in accidents:<br>Car - 497,992<br>Van / Goods 3.5 tonnes mgw or under - 34,160<br>Bus or coach (17 or more pass seats) - 25,878<br>Motorcycle over 500cc - 25,657<br>Goods 7.5 tonnes mgw and over - 17,307</h3>

<h1>20. What are the summary stats for casualties and vehicles?</h1>

In [27]:
df[["Number_of_Casualties", "Number_of_Vehicles"]].describe()

Number_of_Casualties  Number_of_Vehicles
count         660679.000000       660679.000000
mean               1.357040            1.831255
std                0.824847            0.715269
min                1.000000            1.000000
25%                1.000000            1.000000
50%                1.000000            2.000000
75%                1.000000            2.000000
max               68.000000           32.000000

<h3>Insight : This shows the smallest, biggest, and average numbers for both casualties and vehicles in the accidents.</h3>

<h1>21. How many accidents happened each year?</h1>

In [28]:
df["Year"].value_counts().head(5)

Year
2019    182115
2020    170591
2021    163554
2022    144419
Name: count, dtype: int64

<h3>Insight : Every year have many accidents, and the year 2019 has the highest counts.</h3>

<h1>22. Which month has the most accidents?</h1>

In [29]:
df["Month"].mode()[0]

np.int32(11)

<h3>Insight : The most common month for accidents is November (month 11).</h3>

<h1>23. Which day of the week has the most accidents?</h1>

In [30]:
df["DayOfWeek"].mode()[0]

np.int32(5)

<h3>Insight : The most common accident day is Friday (Day 5).</h3>

<h1>24. On average, how many accidents happen in a month?</h1>

In [31]:
df.groupby("Month")["Index"].count().mean()

np.float64(55056.583333333336)

<h3>Insight : On average, about 55,057 accidents happen in a month.</h3>

<h1>25. Which 3 days of the month have the most accidents?</h1>

In [32]:
df["Day"].value_counts().head(3)

Day
1     22606
12    22536
11    22503
Name: count, dtype: int64

<h3>Insight : Accidents happen a lot on the 1st, 11th, and 12th day of each month.<br>Day 1 of the month had 22,606 accidents (most).
<br>Day 12 had 22,536 accidents.<br>Day 11 had 22,503 accidents.</h3>

In [35]:
df['District Area'].unique()

array(['Kensington and Chelsea', 'Westminster', 'Richmond upon Thames',
       'Hammersmith and Fulham', 'Hounslow', 'Tower Hamlets',
       'City of London', 'Southwark', 'Camden', 'Hackney', 'Islington',
       'Barnet', 'Brent', 'Haringey', 'Merton', 'Ealing', 'Enfield',
       'Greenwich', 'Newham', 'London Airport (Heathrow)', 'Hillingdon',
       'Waltham Forest', 'Redbridge', 'Barking and Dagenham', 'Bromley',
       'Havering', 'Croydon', 'Lambeth', 'Wandsworth', 'Sutton', 'Bexley',
       'Lewisham', 'Harrow', 'Kingston upon Thames', 'Barrow-in-Furness',
       'South Lakeland', 'Carlisle', 'Eden', 'Allerdale', 'Copeland',
       'Fylde', 'Blackpool', 'Wyre', 'Lancaster', 'Chorley',
       'West Lancashire', 'South Ribble', 'Preston',
       'Blackburn with Darwen', 'Hyndburn', 'Ribble Valley', 'Burnley',
       'Pendle', 'Rossendale', 'Wirral', 'Liverpool', 'Sefton',
       'Knowsley', 'St. Helens', 'Manchester', 'Salford', 'Tameside',
       'Stockport', 'Bolton', 'Wigan', '

In [37]:
northumberland = df[df['District Area'] == 'Northumberland']

In [48]:
northumberland

AttributeError: 'str' object has no attribute 'dtype'

In [40]:
county_durham = df[df['District Area'] == 'County Durham']
county_durham

Index Accident_Severity Accident Date   Latitude  \
399187  200911ND07999            Slight    2021-04-02  54.752380   
399188  200911ND08009            Slight    2021-04-03  54.765338   
399189  200911ND08109            Slight    2021-04-05  54.762584   
399190  200911ND08289            Slight    2021-04-12  54.729756   
399192  200911ND08599            Slight    2021-04-06  54.805789   
...               ...               ...           ...        ...   
562762  201011SL28650            Slight    2022-11-29  54.704466   
562763  201011SL29160            Slight    2022-12-06  54.705043   
562764  201011SL29590            Slight    2022-12-13  54.703521   
562765  201011SL30490            Slight    2022-12-23  54.681451   
562766  201011SL31500            Slight    2022-12-18  54.661023   

             Light_Conditions  District Area  Longitude  Number_of_Casualties  \
399187               Daylight  County Durham  -1.472792                     1   
399188               Daylight  County Durham  -1.520027                     2   
399189               Daylight  County Durham  -1.463335                     1   
399190               Daylight  County Durham  -1.521535                     1   
399192               Daylight  County Durham  -1.522815                     1   
...                       ...            ...        ...                   ...   
562762  Darkness - lights lit  County Durham  -1.611687                     2   
562763               Daylight  County Durham  -1.595852                     1   
562764               Daylight  County Durham  -1.597729                     1   
562765  Darkness - lights lit  County Durham  -1.532960                     1   
562766  Darkness - lights lit  County Durham  -1.549317                     1   

        Number_of_Vehicles Road_Surface_Conditions           Road_Type  \
399187                   2                     Dry  Single carriageway   
399188                   2                     Dry  Single carriageway   
399189                   1                     Dry  Single carriageway   
399190                   2                     Dry           Slip road   
399192                   2                     Dry    Dual carriageway   
...                    ...                     ...                 ...   
562762                   2                    Snow  Single carriageway   
562763                   1                     Dry  Single carriageway   
562764                   1                    Snow  Single carriageway   
562765                   2                    Snow  Single carriageway   
562766                   2             Wet or damp  Single carriageway   

       Urban_or_Rural_Area     Weather_Conditions                Vehicle_Type  \
399187               Rural     Fine no high winds                         Car   
399188               Rural     Fine no high winds                         Car   
399189               Rural     Fine no high winds                         Car   
399190               Rural     Fine no high winds                         Car   
399192               Rural     Fine no high winds  Motorcycle 125cc and under   
...                    ...                    ...                         ...   
562762               Urban  Snowing no high winds                         Car   
562763               Urban     Fine no high winds                         Car   
562764               Urban     Fine no high winds                         Car   
562765               Urban  Snowing no high winds                         Car   
562766               Rural     Fine no high winds                         Car   

        Year  Month  Day  DayOfWeek  
399187  2021      4    2          4  
399188  2021      4    3          5  
399189  2021      4    5          0  
399190  2021      4   12          0  
399192  2021      4    6          1  
...      ...    ...  ...        ...  
562762  2022     11   29          1  
562763  2022     12    6          1  
562764  2022     12   

In [41]:
shropshire = df[df['District Area'] == 'Shropshire']
shropshire

Index Accident_Severity Accident Date   Latitude  \
427614  200922C903610           Serious    2021-07-04  52.313976   
428500  200922D905319            Slight    2021-09-24  52.538912   
429290  200922F000035            Slight    2021-12-26  52.378853   
429291  200922F000061            Slight    2021-12-30  52.713960   
429292  200922F000077            Slight    2021-12-19  52.707812   
...               ...               ...           ...        ...   
589747  201022G005879            Slight    2022-11-26  52.537143   
589751  201022G005918            Slight    2022-11-25  52.418938   
589763  201022G006041             Fatal    2022-12-04  52.551668   
589770  201022G006158            Slight    2022-12-06  52.545487   
589796  201022G100246            Slight    2022-12-07  52.546410   

               Light_Conditions District Area  Longitude  \
427614                 Daylight    Shropshire  -2.594808   
428500    Darkness - lights lit    Shropshire  -2.807656   
429290  Darkness - lights unlit    Shropshire  -2.484029   
429291                 Daylight    Shropshire  -2.819323   
429292                 Daylight    Shropshire  -2.757042   
...                         ...           ...        ...   
589747                 Daylight    Shropshire  -2.445665   
589751   Darkness - no lighting    Shropshire  -2.416534   
589763                 Daylight    Shropshire  -2.500381   
589770   Darkness - no lighting    Shropshire  -2.399888   
589796   Darkness - no lighting    Shropshire  -2.262752   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
427614                     1                   1                     Dry   
428500                     1                   2                     Dry   
429290                     1                   1            Frost or ice   
429291                     4                   2             Wet or damp   
429292                     2                   1                     Dry   
...                      ...                 ...                     ...   
589747                     2                   2                     Dry   
589751                     1                   2            Frost or ice   
589763                     3                   2            Frost or ice   
589770                     2                   2            Frost or ice   
589796                     1                   1            Frost or ice   

                 Road_Type Urban_or_Rural_Area  Weather_Conditions  \
427614  Single carriageway               Rural  Fine no high winds   
428500  Single carriageway               Rural  Fine no high winds   
429290  Single carriageway               Rural               Other   
429291    Dual carriageway               Rural  Fine no high winds   
429292  Single carriageway               Urban  Fine no high winds   
...                    ...                 ...                 ...   
589747  Single carriageway               Rural  Fine no high winds   
589751  Single carriageway               Rural  Fine no high winds   
589763  Single carriageway               Rural  Fine no high winds   
589770  Single carriageway               Rural         Fog or mist   
589796  Single carriageway               Rural  Fine no high winds   

                                Vehicle_Type  Year  Month  Day  DayOfWeek  
427614                                   Car  2021      7    4          6  
428500                                   Car  2021      9   24          4  
429290                                   Car  2021     12   26          6  
429291                                   Car  2021     12   30          3  
429292                                   Car  2021     12   19          6  
...                                      ...   ...    ...  ...        ...  
589747                                   Car  2022     11   26          5  
589751                                   Car  2022     11   25          4  
589763                                  

In [42]:
central_dedfordshire = df[df['District Area'] == 'Central Bedfordshire']
central_dedfordshire

Index Accident_Severity Accident Date   Latitude  \
451328  2009400B00105            Slight    2021-12-20  52.071772   
451370  2009400B00492            Slight    2021-12-17  52.071772   
451434  2009400B00970            Slight    2021-04-04  52.085321   
451437  2009400B00992            Slight    2021-04-06  52.126788   
451438  2009400B01004           Serious    2021-04-13  52.139094   
...               ...               ...           ...        ...   
611289  2010400D03772            Slight    2022-12-19  52.015115   
611290  2010400D03795            Slight    2022-12-29  51.984249   
611291  2010400D03803            Slight    2022-12-22  51.931456   
611292  2010400D07161            Slight    2022-06-09  51.963476   
611293  2010400T02138           Serious    2022-11-27  52.011574   

              Light_Conditions         District Area  Longitude  \
451328                Daylight  Central Bedfordshire  -0.243149   
451370   Darkness - lights lit  Central Bedfordshire  -0.243149   
451434                Daylight  Central Bedfordshire  -0.265533   
451437                Daylight  Central Bedfordshire  -0.303660   
451438  Darkness - no lighting  Central Bedfordshire  -0.321605   
...                        ...                   ...        ...   
611289  Darkness - no lighting  Central Bedfordshire  -0.297687   
611290  Darkness - no lighting  Central Bedfordshire  -0.634087   
611291   Darkness - lights lit  Central Bedfordshire  -0.587256   
611292                Daylight  Central Bedfordshire  -0.522204   
611293                Daylight  Central Bedfordshire  -0.204562   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
451328                     3                   1            Frost or ice   
451370                     1                   1             Wet or damp   
451434                     1                   1                     Dry   
451437                     1                   2                     Dry   
451438                     1                   1                     Dry   
...                      ...                 ...                     ...   
611289                     1                   1            Frost or ice   
611290                     1                   2            Frost or ice   
611291                     1                   2            Frost or ice   
611292                     1                   2             Wet or damp   
611293                     3                   1                    Snow   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
451328          Roundabout               Rural                  Other   
451370          Roundabout               Rural                  Other   
451434  Single carriageway               Urban     Fine no high winds   
451437  Single carriageway               Rural     Fine no high winds   
451438  Single carriageway               Rural     Fine no high winds   
...                    ...                 ...                    ...   
611289  Single carriageway               Urban                  Other   
611290  Single carriageway               Rural     Fine no high winds   
611291  Single carriageway               Rural  Snowing no high winds   
611292    Dual carriageway               Rural  Raining no high winds   
611293           Slip road               Rural  Snowing no high winds   

                               Vehicle_Type  Year  Month  Day  DayOfWeek  
451328                                  Car  2021     12   20          0  
451370                                  Car  2021     12   17          4  
451434                                  Car  2021      4    4          6  
451437                Motorcycle over 500cc  2021      4    6          1  
451438                                  Car  2021      4   13          1  
...                                     ...   ...    ...  ...        ...  
611289                                  Car  2022     12   19          0  
6112

In [43]:
cornwall = df[df['District Area'] == 'Cornwall']
cornwall

Index Accident_Severity Accident Date   Latitude  \
484274  200950AC2B004            Slight    2021-04-06  50.180885   
484275  200950AC2B005           Serious    2021-04-10  50.211390   
484276  200950AC2B006           Serious    2021-04-30  50.208114   
484277  200950AC2B009            Slight    2021-06-05  50.212727   
484278  200950AC2B011            Slight    2021-07-02  50.194520   
...               ...               ...           ...        ...   
644168  201050GW3R007            Slight    2022-08-07  50.334243   
644169  201050GW3R008           Serious    2022-09-07  50.343975   
644171  201050GW3R010           Serious    2022-11-17  50.334272   
644172  201050GW3R011            Slight    2022-12-08  50.343758   
644778  201050JS2J001            Slight    2022-02-07  50.390923   

              Light_Conditions District Area  Longitude  Number_of_Casualties  \
484274                Daylight      Cornwall  -5.281875                     1   
484275   Darkness - lights lit      Cornwall  -5.291119                     1   
484276                Daylight      Cornwall  -5.292436                     1   
484277                Daylight      Cornwall  -5.288407                     1   
484278  Darkness - no lighting      Cornwall  -5.308597                     3   
...                        ...           ...        ...                   ...   
644168                Daylight      Cornwall  -4.636660                     1   
644169                Daylight      Cornwall  -4.671356                     1   
644171                Daylight      Cornwall  -4.639331                     1   
644172                Daylight      Cornwall  -4.661223                     2   
644778                Daylight      Cornwall  -4.421152                     1   

        Number_of_Vehicles Road_Surface_Conditions           Road_Type  \
484274                   2             Wet or damp  Single carriageway   
484275                   2                     Dry  Single carriageway   
484276                   1                     Dry      One way street   
484277                   1                     Dry  Single carriageway   
484278                   1             Wet or damp  Single carriageway   
...                    ...                     ...                 ...   
644168                   1             Wet or damp  Single carriageway   
644169                   2                     Dry  Single carriageway   
644171                   2             Wet or damp      One way street   
644172                   2                     Dry  Single carriageway   
644778                   1             Wet or damp  Single carriageway   

       Urban_or_Rural_Area     Weather_Conditions  \
484274               Rural     Fine no high winds   
484275               Urban     Fine no high winds   
484276               Urban     Fine no high winds   
484277               Urban     Fine no high winds   
484278               Rural  Raining no high winds   
...                    ...                    ...   
644168               Rural     Fine no high winds   
644169               Rural     Fine no high winds   
644171               Rural     Fine no high winds   
644172               Rural     Fine no high winds   
644778               Rural     Fine no high winds   

                               Vehicle_Type  Year  Month  Day  DayOfWeek  
484274                                  Car  2021      4    6          1  
484275                                  Car  2021      4   10          5  
484276                                  Car  2021      4   30          4  
484277                                  Car  2021      6    5          5  
484278  Van / Goods 3.5 tonnes mgw or under  2021      7    2          4  
...                                     ...   ...    ...  ...        ...  
644168                                  Car  2022      8    7          6  
644169                        Other vehicle  2022      9    7          2  
644171                   

In [44]:
wiltshire = df[df['District Area'] == 'Wiltshire']
wiltshire

Index Accident_Severity Accident Date   Latitude  \
494440  200954DE44509            Slight    2021-12-10  51.630415   
494600  200954EA14709            Slight    2021-04-22  51.346875   
494601  200954EA17009            Slight    2021-05-16  51.358987   
494602  200954EA20609            Slight    2021-06-15  51.360872   
494603  200954EA20709            Slight    2021-06-14  51.343171   
...               ...               ...           ...        ...   
651580  201054MB04410            Slight    2022-09-27  51.548912   
651582  201054MB04610            Slight    2022-10-03  51.517428   
651583  201054MB05410            Slight    2022-11-28  51.509992   
651585  201054MB06010            Slight    2022-12-05  51.545756   
651586  201054MB06110            Slight    2022-12-04  51.507968   

                   Light_Conditions District Area  Longitude  \
494440                     Daylight     Wiltshire  -1.825006   
494600                     Daylight     Wiltshire  -2.243486   
494601                     Daylight     Wiltshire  -2.256189   
494602                     Daylight     Wiltshire  -2.257493   
494603                     Daylight     Wiltshire  -2.251650   
...                             ...           ...        ...   
651580                     Daylight     Wiltshire  -1.863104   
651582       Darkness - no lighting     Wiltshire  -2.253186   
651583  Darkness - lighting unknown     Wiltshire  -2.191616   
651585       Darkness - no lighting     Wiltshire  -1.855470   
651586       Darkness - no lighting     Wiltshire  -1.608908   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
494440                     1                   2             Wet or damp   
494600                     1                   2                     Dry   
494601                     3                   2                     Dry   
494602                     1                   2                     Dry   
494603                     1                   2                     Dry   
...                      ...                 ...                     ...   
651580                     1                   2             Wet or damp   
651582                     2                   5                     Dry   
651583                     1                   1            Frost or ice   
651585                     1                   2            Frost or ice   
651586                     2                   2            Frost or ice   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
494440    Dual carriageway               Rural     Fine no high winds   
494600  Single carriageway               Rural     Fine no high winds   
494601  Single carriageway               Rural     Fine no high winds   
494602  Single carriageway               Rural     Fine no high winds   
494603  Single carriageway               Rural     Fine no high winds   
...                    ...                 ...                    ...   
651580    Dual carriageway               Rural  Raining no high winds   
651582    Dual carriageway               Rural            Fog or mist   
651583    Dual carriageway               Rural     Fine no high winds   
651585    Dual carriageway               Rural     Fine no high winds   
651586    Dual carriageway               Rural            Fog or mist   

                                Vehicle_Type  Year  Month  Day  DayOfWeek  
494440                                   Car  2021     12   10          4  
494600                                   Car  2021      4   22          3  
494601                                   Car  2021      5   16          6  
494602  Bus or coach (17 or more pass seats)  2021      6   15          1  
494603                                   Car  2021      6   14          0  
...                                      ...   ...    ...  ...        ...  
651580                                   Car  2022      9   27          1  
651582   Van / Goods 3.5 tonnes 